bears/uploadTool.py

```diff
@@ -0,0 +1,84 @@
+    if not os.path.exists('upload'):
+        os.mkdir('upload')
+
+    bears_full_filenames = glob.glob('**/*Bear.py')
+
+    for bear_file_name in bears_full_filenames:
+        file_name = bear_file_name
+        bear_name = re.match(r'.+/(.*)\.py', bear_file_name).group(1)
+        bear_object = list(iimport_objects(file_name, attributes='kind',
+                                           local=True))
+        if not os.path.exists(os.path.join('upload', bear_name)):
+            os.mkdir(os.path.join('upload', bear_name))
+        if not os.path.exists(os.path.join('upload', bear_name, bear_name)):
+            os.mkdir(os.path.join('upload', bear_name, bear_name))
+        if not os.path.exists(os.path.join('upload', bear_name, bear_name,
+                                           '__init__.py')):
+            open(os.path.join('upload', bear_name, bear_name, '__init__.py'),
+                 'w').close()
+        shutil.copyfile(file_name, os.path.join('upload', bear_name, bear_name,
+                                                bear_name + '.py'))
+
+        with open(os.path.join('upload', bear_name, 'setup.py'), "w") as setup:
+
+            # set env
+            setup.write('#!/usr/bin/env python3\n')
+
+            # write imports
+            setup.write(IMPORTS + '\n\n')
+
+            # encoding UTF-8
+            setup.write(ENCODING + '\n\n')
+
+            # setuptools
+            setup.write('if __name__== "__main__":\n')
+            setup.write("\tsetup(name=" + "'" + bear_name + "',\n")
+            setup.write('\t\t  version="0.1.30",\n')
+            setup.write("\t\t  description='The " +
+                        bear_name + " bear for coala (Code")
+            setup.write(" Analysis Application)',\n")
+            setup.write("\t\t  authors=" + str(bear_object[0].AUTHORS) + ",\n")
+            setup.write("\t\t  authors_emails="
+                        + str(bear_object[0].AUTHORS_EMAILS) + ",\n")
+            setup.write("\t\t  maintainers=" + str(bear_object[0].maintainers)
+                        + ",\n")
+            setup.write("\t\t  maintainers_emails="
+                        + str(bear_object[0].maintainers_emails) + ",\n")
+            setup.write("\t\t  platforms='" + str(bear_object[0].PLATFORMS)
+                        + "',\n")
+            setup.write("\t\t  license='" +
+                        str(bear_object[0].LICENSE) + "',\n")
+            setup.write('\t\t  classifiers=[')
+            for classifier in CLASSIFIERS:
+                setup.write('\n\t\t\t"' + classifier + '",')
+            setup.write('])')
+
+            # upload everything
+
+            os.chdir(os.path.join('upload', bear_name))
+            # subprocess.call(['python', 'setup.py', 'register', '-r',
+            #                  'pypitest'])
+            # subprocess.call(['python', 'setup.py', 'sdist', 'upload', '-r',
+            #                  'pypitest'])
+            os.chdir(os.path.join('..', '..'))
```

Comment:
this function is way too long! Which also results in it being untested :) make it shorter! Move out stuff into own functions. Can you use templates just like @mr-karan did? I feel like that would make things way easier.